cmake/Modules/BuildSWIG.cmake

```diff
@@ -0,0 +1,22 @@
+include(ExternalProject)
```

Comment:
We originally had swig set up as a developer-only dependency, as it's really only needed to autogenerate some source files -- no need to add another dependency for users. (This came at the cost of keeping autogenerated files in version control, but that wasn't a big problem.) Perhaps CMake makes it easy enough to install that we don't need to worry about this any more? Let's see how testing goes, I guess.

Reply:
Yep, sure, but in all my tests, swig compiled fine when missing from a system. The benefit is that it becomes completely transparent, the interface is automatically regenerated and user/developers don't even have to think about it.